src/profile/v100/entity/display_name.rs

```diff
@@ -35,16 +44,11 @@ impl DisplayName {
     pub const MAX_LEN: usize = 30;
 
     pub fn new(value: impl AsRef<str>) -> Result<Self> {
-        let value = value.as_ref().trim().to_string();
+        let mut value = value.as_ref().trim().to_string();
         if value.is_empty() {
             return Err(CommonError::InvalidDisplayNameEmpty);
         }
-        if value.len() > Self::MAX_LEN {
-            return Err(CommonError::InvalidDisplayNameTooLong {
-                expected: Self::MAX_LEN as u64,
-                found: value.len() as u64,
-            });
-        }
+        value.truncate(Self::MAX_LEN);
```

Comment:
use the same you used in recent PR? `.chars().take(Self::MAX_LEN).collect()` instead? which seemed safe.?